api/src/main/java/org/openmrs/api/impl/PersonServiceImpl.java

```diff
@@ -296,44 +296,60 @@ public void purgePerson(Person person) throws APIException {
 	 */
 	@Override
 	public Person savePerson(Person person) throws APIException {
+		setPreferredPersonName(person);
+		setPreferredPersonAddress(person);
+		return dao.savePerson(person);
+	}
+
+	/**
+	 * Set preferred name of person
+	 *
+	 * @param person Person whose preferred name is to be set
+	 */
```

Comment:
@MalshaL same comment I gave on your other PR applies. please remove the javadocs you added